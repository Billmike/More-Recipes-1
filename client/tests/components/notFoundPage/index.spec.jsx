import React from 'react';

import NotFoundPage from '../../../components/notFoundPage';

describe('NotFoundPage component', () => {
  const props = {
    history: {
      goBack: jest.fn(),
    },
  };
  it('should render component', () => {
    const wrapper = shallow(<NotFoundPage {...props} />);

    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
  it('should call go back function on click', () => {
    const wrapper = shallow(<NotFoundPage {...props} />);
    wrapper.find('a').simulate('click');
    expect(props.history.goBack).toBeCalled();
  });
});
