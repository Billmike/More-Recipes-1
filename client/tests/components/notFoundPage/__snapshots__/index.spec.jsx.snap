// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage component should render component 1`] = `
<div
  className="row"
>
  <nav>
    <div
      className="nav-wrapper deep-orange darken-4"
    >
      <Link
        className="brand-logo hide-on-small-only"
        id="logo"
        replace={false}
        to="/"
      >
        More-Recipes
      </Link>
    </div>
  </nav>
  <div
    className="row center-align"
  >
    <div
      className="not-found"
    >
      <p
        className="status-number"
      >
        404
      </p>
      <p
        className="page-data"
      >
        PAGE NOT FOUND
      </p>
      <i
        className="material-icons large"
      >
        error
      </i>
    </div>
    <span>
      <a
        href="#!"
        onClick={[MockFunction]}
      >
        Go back 
      </a>
      <span>
        |
      </span>
      <Link
        replace={false}
        to="/"
      >
         Go home
      </Link>
    </span>
  </div>
</div>
`;
